extensions/ql-vscode/src/remote-queries/analyses-results-manager.ts

```diff
@@ -58,7 +59,12 @@ export class AnalysesResultsManager {
       const nwos = batch.map(a => a.nwo).join(', ');
       void this.logger.log(`Downloading batch ${Math.floor(i / batchSize) + 1} of ${numOfBatches} (${nwos})`);
 
-      await Promise.all(batchTasks);
+      const taskResults = await Promise.allSettled(batchTasks);
+      const failedTasks = taskResults.filter(x => x.status === 'rejected') as Array<PromiseRejectedResult>;
+      if (failedTasks.length > 0) {
+        const failures = failedTasks.map(t => t.reason.message).concat(os.EOL);
+        throw Error(failures.join(os.EOL));
+      }
```

Comment:
So, this leaves us in a state where batches are downloaded until there is a batch with at least one error and then all further downloading stops. I think it should be possible to continue attempting to download even after a single error occurs.

If you accumulate failures in a list within the loop (maybe you can log to the output view so users can see errors as they happen), and then you can throw after _all_ the requests are settled. This way, users can see errors as they occur (if they want), they get a summary of all errors at the end, and a single failure doesn't prevent further downloads from happening.

Reply:
That's a good point, sorry I missed it yesterday. Updated, hopefully there now 🤞